### PR TITLE
chore: extract download simple auth as a single action

### PR DESCRIPTION
**Pull request by wenytang-ms**

### CODE REVIEW

**Review by qinezh — APPROVED** [review]

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -75,18 +75,7 @@ jobs:
...
-          max_attempts: 10
-          retry_on: error
-          shell: pwsh
-          command: |
-            $version=Get-Content packages/fx-core/templates/plugins/resource/simpleauth/version.txt
-            $tag = "simpleauth@"+$version
-            $fileName="Microsoft.TeamsFx.SimpleAuth_$version.zip"
-            $url = "https://github.com/OfficeDev/TeamsFx/releases/download/"+$tag+"/"+$fileName
-            Invoke-WebRequest $url -OutFile packages/fx-core/templates/plugins/resource/simpleauth/SimpleAuth.zip
+        uses: ./.github/workflows/action/download-simple-auth.yml
```
Comment (qinezh) [inline .github/workflows/cd.yml:78]: CI failed:

> Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under '/home/runner/work/TeamsFx/TeamsFx/.github/workflows/action/download-simple-auth.yml'. Did you forget to run actions/checkout before running your local action?

**Review by qinezh — CHANGES_REQUESTED** [review]

**Review by adashen — APPROVED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (wenytang-ms) [reply .github/workflows/cd.yml]: yes, processing.

**Review by LongOddCode — APPROVED** [review]

**Review by qinezh — APPROVED** [review]

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -93,11 +93,7 @@ jobs:
           retry_on: error
           shell: pwsh
           command: |
-            $version=Get-Content packages/fx-core/templates/plugins/resource/simpleauth/version.txt
-            $tag = "simpleauth@"+$version
-            $fileName="Microsoft.TeamsFx.SimpleAuth_$version.zip"
-            $url = "https://github.com/OfficeDev/TeamsFx/releases/download/"+$tag+"/"+$fileName
-            Invoke-WebRequest $url -OutFile packages/fx-core/templates/plugins/resource/simpleauth/SimpleAuth.zip
+            ./.github/scripts/downloadSimpleAuth.ps1
```
Comment (adashen) [inline .github/workflows/cd.yml:96]: will powershell fail on linux agent?

Comment (wenytang-ms) [reply .github/workflows/cd.yml:96]: About GitHub-hosted runners - GitHub Docs
virtual-environments/Ubuntu2004-README.md at main · actions/virtual-environments (github.com)
powershell is the default shell installed on images from github

**Pull request merged**